pyccl/halos/massdef.py

```diff
@@ -0,0 +1,247 @@
+        """ Translates a halo mass into a radius
+            M =  (4 * pi / 3) * rho_X * Delta * R^3
+
+        Args:
+            cosmo (:obj:`Cosmology`): A Cosmology object.
+            M (float or array_like): halo mass in units of M_sun.
+            a (float): scale factor.
+
+        Returns:
+            float or array_like: halo radius in units of Mpc (physical, not
+                comoving).
+        """
+        Delta = self.get_Delta(cosmo, a)
+        return (M / (4.18879020479 * Delta *
+                     rho_x(cosmo, a, self.rho_type)))**(1./3.)
+
+    def get_concentration(self, cosmo, M, a):
+        """ Returns concentration for this mass definition.
+
+        Args:
+            cosmo (:obj:`Cosmology`): A Cosmology object.
+            M (float or array_like): halo mass in units of M_sun.
+            a (float): scale factor.
+
+        Returns:
+            float or array_like: halo concentration.
+        """
+        if self.concentration is None:
+            raise RuntimeError("This mass definition doesn't have "
+                               "an associated c(M) relation")
+        else:
+            return self.concentration.get_concentration(cosmo, M, a)
+
+    def translate_mass(self, cosmo, M, a, m_def_other):
+        """ Translate halo mass in this definition into another definition
+
+        Args:
+            cosmo (:obj:`Cosmology`): A Cosmology object.
+            M (float or array_like): halo mass in units of M_sun.
+            a (float): scale factor.
+            m_def_other (:obj:`MassDef`): another mass definition.
+
+        Returns:
+            float or array_like: halo masses in new definition.
+        """
+        if self == m_def_other:
+            return M
+        else:
+            if self.concentration is None:
+                raise RuntimeError("This mass definition doesn't have "
+                                   "an associated c(M) relation")
+            else:
+                om_this = omega_x(cosmo, a, self.rho_type)
+                D_this = self.get_Delta(cosmo, a) * om_this
+                c_this = self.get_concentration(cosmo, M, a)
+                R_this = self.get_radius(cosmo, M, a)
+                om_new = omega_x(cosmo, a, m_def_other.rho_type)
+                D_new = m_def_other.get_Delta(cosmo, a) * om_new
+                c_new = convert_concentration_py(cosmo, c_this, D_this, D_new)
+                R_new = c_new * R_this / c_this
+                return m_def_other.get_mass(cosmo, R_new, a)
+
+
+class MassDef200mat(MassDef):
+    """`MassDef` class for the mass definition with Delta=200 times the matter
+    density.
+
+    Args:
+        c_m (string): concentration-mass relation.
+    """
+    def __init__(self, c_m='Duffy08'):
+        super(MassDef200mat, self).__init__(200,
+                                            'matter',
+                                            c_m_relation=c_m)
+
+
+class MassDef200crit(MassDef):
```

Comment:
`200crit` or `200c`? Again the second is more common.

Reply:
Done